File: rxbinding/src/main/java/com/forceson/rxbinding/widget/RxRadioGroup.java
package com.forceson.rxbinding.widget;


import android.widget.RadioGroup;

import io.reactivex.Observable;
import io.reactivex.functions.Consumer;

/**
 * Created by son on 2020-01-25.
 */
public class RxRadioGroup {
    public static Observable<Integer> checkedChanges(RadioGroup view) {
        return new RadioGroupCheckedChangeObservable(view);
    }

    public static Consumer<Integer> setChecked(final RadioGroup view) {
        return new Consumer<Integer>() {
            @Override
            public void accept(Integer value) {
                if (value == -1) {
                    view.clearCheck();
                } else {
                    view.check(value);
                }
            }
        };
    }
}

File: rxbinding-kotlin/src/main/java/com/forceson/rxbinding/kotlin/widget/RxRadioGroup.kt
package com.forceson.rxbinding.kotlin.widget

import android.widget.RadioGroup
import com.forceson.rxbinding.widget.RxRadioGroup
import rx.Observable
import rx.functions.Action1

/**
 * Created by son on 2020-02-10.
 */

fun RadioGroup.checkedChanges(): Observable<Int> = RxRadioGroup.checkedChanges(this)

fun RadioGroup.checkedChangeEvents(): Observable<RadioGroupCheckedChangeEvent> = RxRadioGroup.checkedChangeEvents(this)

fun RadioGroup.checked(): Action1<in Int> = RxRadioGroup.setChecked(this)